### PR TITLE
Add retry-logic to create groups + other improvements

### PR DESCRIPTION
**Pull request by ekinanp**

### CODE REVIEW

File: classifier_load_testing/create_groups.rb [hunk 1]
```diff
@@ -19,52 +19,51 @@
...
-num_nodes.to_i.times do |node|
-  node = node + start_at
-  node_name = "node#{node}"
-  if (node % 100) == 0
-    puts(node_name)
+overallElapsedTime = time_block do
+  # Common helper that removes the "with_retry", "time_block" clutter.
+  # Helper returns how long it took to create the group.
+  create_group = lambda do |group|
+    retryDelay = per_node_delay.zero? ? 5 : 5 * per_node_delay
```
Comment (ekinanp) [inline classifier_load_testing/create_groups.rb:26]: `3` is an arbitrary # of retries. retryDelay is also arbitrary. Figured 5 * node_delay should give API more time to settle down.

File: classifier_load_testing/create_groups.rb [hunk 2]
```diff
@@ -4,56 +4,66 @@
...
+    with_retry("Create the #{group['name']} group", 3, retryDelay) do
+      time_block do
+        sleep 1
```
Comment (RandomNoun7) [inline classifier_load_testing/create_groups.rb]: If the sleep is inside the `time_block` does that get added to the elapsed time of the api call?

Comment (ekinanp) [reply classifier_load_testing/create_groups.rb]: Oh shit I forgot to remove that. That was me locally testing the elapsed time. Thanks for catching

Comment (ekinanp) [reply classifier_load_testing/create_groups.rb]: Updated.

File: classifier_load_testing/create_groups.rb (hunk 1 above)
Comment (RandomNoun7) [reply classifier_load_testing/create_groups.rb:26]: 👍

**Review by RandomNoun7 — APPROVED** [review]

**Review by RandomNoun7 — APPROVED** [review]

**Pull request merged**